test/test_buffer.cpp

```diff
@@ -139,6 +139,11 @@ BOOST_AUTO_TEST_CASE(destructor_templated_callback)
         boost::compute::buffer buf(context, 128);
         buf.set_destructor_callback(boost::bind(destructor_templated_callback_function, &invoked));
     }
+
+    // NVIDIA calls the callback in a background thread and so the check is executed
+    // before the callback function is called
+    sleep(1);
```

Comment:
The way I've dealt with this before is to use a `mutex` and `condition_variable::wait_for()` to wait for the event to complete with a finte time-out. See the `event_callback` test in [test_event.cpp](https://github.com/kylelutz/compute/blob/master/test/test_event.cpp#L44) for an example.